### PR TITLE
Metadata service integration

### PR DESCRIPTION
**Pull request by KennyZeroTech**
Adds the function `compareNftsToMetadataService` that uses the metadata-service to compare against a given array of Nfts

This can be tested by running the normal test code without specifying an "original" file

### CODE REVIEW

File: src/comparers/compareNFTs.ts [hunk 1]
```diff
@@ -11,7 +11,10 @@ import {
   NftDiff,
   NftFileData,
```
Comment (JamesEarle) [inline src/comparers/compareNFTs.ts:12]: Bit of a nitpick, but there is inconsistency with the use of NFTs, Nfts, and nfts.

File: src/comparers/compareNFTs.ts [hunk 2]
```diff
@@ -204,3 +207,40 @@ export const compareNftFileToDatabase = async (
...
+interface PostReturn {
+  warnings: string;
+  metadata: Map<Metadata>;
+}
+
+export const compareNftsToMetadataService = async (
+  modifiedNfts: NftData[],
+  logger: Logger
+): Promise<NftBatchDiff> => {
+  const nftDataForDatabase: { domainIds: string[] } = { domainIds: [] };
```
Comment (JamesEarle) [inline src/comparers/compareNFTs.ts:220]: Consider adding some typing here

**Review by JamesEarle — APPROVED** [review]

File: src/comparers/compareNFTs.ts (hunk 2 above)
Comment (KennyZeroTech) [reply src/comparers/compareNFTs.ts:188]: Fixed

File: src/comparers/compareNFTs.ts [hunk 3]
```diff
@@ -178,29 +176,39 @@ export const compareNftFiles = (
   return diff;
 };
 
-export const compareNftGroupToDatabase = async (
-  modifiedNfts: NftData[],
+interface PostReturn {
```
Comment (RyanCross) [inline src/comparers/compareNFTs.ts:179]: This could have a more descriptive name

File: src/comparers/compareNFTs.ts [hunk 4]
```diff
@@ -178,29 +176,39 @@ export const compareNftFiles = (
...
-  const databaseNfts = await getNftsFromDatabase(nftDataForDatabase, logger);
+  const metadataServiceUrl = env
+    .get("METADATA_SERVICE_URL")
+    .default("https://metadata-service-api.azurewebsites.net/api/metadata")
+    .asString();
 
-  const diff = compareNftGroups(databaseNfts, modifiedNfts, logger);
-  return diff;
-};
+  const serviceData: PostReturn = (
```
Comment (RyanCross) [inline src/comparers/compareNFTs.ts:199]: Should there be some retry or error handling logic around this request?

**Review by Will-Hyde — APPROVED** [review]

**Comment by RyanCross** [comment]
Please address those circleci build errors before merging.

File: src/comparers/compareNFTs.ts (hunk 3 above)
Comment (KennyZeroTech) [reply src/comparers/compareNFTs.ts:179]: Fixed and added to types

File: src/comparers/compareNFTs.ts (hunk 4 above)
Comment (KennyZeroTech) [reply src/comparers/compareNFTs.ts:199]: Added error handling for if response failed

**Pull request merged**